oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzurePersistenceManager.java

```diff
@@ -0,0 +1,199 @@
+package org.apache.jackrabbit.oak.segment.azure;
+
+import com.azure.identity.ClientSecretCredential;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.common.policy.RequestRetryOptions;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jackrabbit.oak.segment.azure.util.AzureRequestOptions;
+import org.apache.jackrabbit.oak.segment.azure.util.Environment;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static org.apache.jackrabbit.oak.segment.azure.AzureUtilities.*;
+
+public class AzurePersistenceManager {
+
+    private static final Logger log = LoggerFactory.getLogger(AzurePersistenceManager.class);
+
+    private AzurePersistenceManager() {
+    }
+
+    public static AzurePersistence createAzurePersistenceFrom(@NotNull String accountName, @NotNull String containerName, @NotNull String rootPrefix, @NotNull Environment environment) throws IOException {
```

Comment:
this will be used in multiple places once the "tooling" will be updated to use the new implementation with v 12, eg: https://github.com/ierandra/jackrabbit-oak/blob/18db288fdd1e732ef565e53ab38f37c1e63ba08e/oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/tool/ToolUtils.java#L178